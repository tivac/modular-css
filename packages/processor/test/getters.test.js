"use strict";

const { DepGraph } = require("dependency-graph");

const namer    = require("@modular-css/test-utils/namer.js");
const relative = require("@modular-css/test-utils/relative.js");
const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe("getters", () => {
        let processor;
        
        beforeEach(() => {
            processor = new Processor({
                namer,
            });
        });
        
        describe(".file", () => {
            it("should return all the files that have been added", async () => {
                await processor.file("./packages/processor/test/specimens/start.css");
                await processor.file("./packages/processor/test/specimens/local.css");

                expect(
                    relative(Object.keys(processor.files))
                )
                .toMatchSnapshot();
            });
        });

        describe(".options", () => {
            it("should return the merged options object", () =>
                expect(typeof processor.options).toBe("object")
            );
        });

        describe(".graph", () => {
            it("should return the dependency graph for added CSS files", async () => {
                await processor.file("./packages/processor/test/specimens/start.css");
                await processor.file("./packages/processor/test/specimens/local.css");

                expect(processor.graph).toBeInstanceOf(DepGraph);
            });
        });
    });
});
