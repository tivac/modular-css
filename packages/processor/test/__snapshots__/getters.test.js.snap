// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js getters .file should return all the files that have been added 1`] = `
[
  "packages/processor/test/specimens/start.css",
  "packages/processor/test/specimens/local.css",
  "packages/processor/test/specimens/folder/folder.css",
]
`;
